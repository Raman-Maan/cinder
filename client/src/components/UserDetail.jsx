import React from 'react';

import './styles/UserDetail.css';
import genderMale from '../assets/gender-male.svg';
import genderFemale from '../assets/gender-female.svg';
import genderOther from '../assets/gender-other.svg';

class UserDetail extends React.Component {
  getGenderPic() {
    if (!this.props.userDetail || !this.props.userDetail.userName) {
      return;
    }
    if (this.props.userDetail.userGender === 'Male') return genderMale;
    else if (this.props.userDetail.userGender === 'Female') return genderFemale;
    else return genderOther;
  }

  dateFormat() {
    return (
      this.props.userDetail.matchTime.toLocaleDateString() +
      ' ' +
      this.props.userDetail.matchTime.toLocaleTimeString()
    );
  }

  render() {
    if (!this.props.userDetail || !this.props.userDetail.userName) {
      return (
        <div className="UserDetail">
          <div className="empty">
            <h3 className="msg align-self-center">
              No match selected yet. Select one match from the list!
            </h3>
          </div>
        </div>
      );
    }

    return (
      <div className="UserDetail">
        <img
          className="userImage"
          src={this.props.userDetail.userPics}
          alt=""
        />
        <h2>{this.props.userDetail.userName}</h2>
        <div className="userAgeWithGender">
          <span className="userAge">{this.props.userDetail.userAge}</span>
          <img className="userGender" src={this.getGenderPic()} alt="" />
        </div>
        <div>
          <p>
            <span className="titleStyle">Matched:</span>{' '}
            <span>
              {this.props.userDetail.matchTime ? this.dateFormat() : 'Unkown'}{' '}
            </span>
          </p>
        </div>
        <div>
          <span className="titleStyle">Bio:</span>{' '}
          <p className="textAutoWrap">{this.props.userDetail.userBio}</p>
        </div>
      </div>
    );
  }
}

export default UserDetail;
