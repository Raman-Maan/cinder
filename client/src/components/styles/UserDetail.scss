$empty-msg-color: #aaa;

.UserDetail {
  height: 100%;
  width: 100%;

  text-align: center;

  .empty {
    display: inline-block;
    width: 100%;
    height: 100%;

    padding: 1em;

    justify-content: center;

    .msg {
      color: $empty-msg-color;
    }
  }

  .userImage {
    display: inline-block;
    max-width: 300px;
    max-height: 200px;
    width: auto;
    height: auto;
  }

  .userAgeWithGender {
    .userAge {
      display: inline-block;
      margin-right: 30px;
    }

    .userGender {
      display: inline-block;
      max-width: 30px;
      max-height: 20px;
      width: auto;
      height: auto;
    }
  }

  .titleStyle {
    font-weight: bold;
  }

  .textAutoWrap {
    margin: 0 auto;
    max-width: 500px;
    padding: 12px;
  }
}
