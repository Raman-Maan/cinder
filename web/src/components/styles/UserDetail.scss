$white: var(--white);
$gray: var(--gray);
$light: var(--light);
$image-size: 300px;
$border-size: 8px;

$box-shadow-x: 2px;
$box-shadow-y: 4px;
$box-shadow-blur: 10px;
$box-shadow-colour: rgba(0, 0, 0, 0.3);

$hover-shadow-colour: rgba(0, 0, 0, 0.4);
$hover-shadow-y: 6px;

$box-shadow: $box-shadow-x $box-shadow-y $box-shadow-blur $box-shadow-colour;
$hover-box-shadow: $box-shadow-x $hover-shadow-y $box-shadow-blur $box-shadow-colour;

$spacing: 1em 0;

$body-font-size: 20px;

$button-spacing: 5em;

$navbar-height: 62px;
$back-button-height: 5em;
$height-offset: $navbar-height + 2em;

.UserDetail {
  height: calc(100% - #{$height-offset});
  width: 100%;

  .back {
    margin-bottom: $back-button-height;

    button {
      position: absolute;
      top: 0;
      left: 0;
      margin: 1em;
    }
  }

  .image {
    display: flex;
    position: relative;
    justify-content: center;
    align-items: center;

    margin-left: auto;
    margin-right: auto;
    
    height: $image-size;
    width: $image-size;
    border-radius: 50%;

    border: $border-size solid $white;
    box-shadow: $box-shadow;

    transition: box-shadow 150ms ease-out;

    img {
      width: calc(100% - #{$border-size});
      height: calc(100% - #{$border-size});
      border-radius: 50%;
      background-color: $light;
      object-fit: scale-down;
    }

    &:hover {
      box-shadow: $hover-box-shadow;
    }
  }

  .header {
    font-weight: bold;
    text-align: center;
    margin: $spacing;
  }

  .details {
    font-size: $body-font-size;

    margin-left: auto;
    margin-right: auto;

    & > .detail {
      margin: $spacing;
    }
  }

  .matchBtns {
    display: flex;
    justify-content: center;
    align-items: center;

    & > button {
      margin: 0 $button-spacing;
    }
  }

  .matchDate {
    position: absolute;
    bottom: 0;
    left: 50%;
    margin-bottom: 1em;
    color: $gray;
  }
}

.divider {
  position: relative;
  width: 100%;
  text-align: center;

  p {
    display: inline;
    background-color: $white;
    padding: 0.5em;
    font-weight: bold;
  }

  &::after {
    content: '';
    position: absolute;
    top: 50%;
    left: 0;
    width: 100%;
    height: 1px;
    background-color: $gray;
    z-index: -1;
  }
}
