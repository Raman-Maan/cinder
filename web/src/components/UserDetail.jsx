import React from 'react';
import { Button } from 'reactstrap';

import './styles/UserDetail.css';

class UserDetail extends React.Component {
  formatMatchDate() {
    return new Date(this.props.matchDate).toISOString().split('T')[0];
  }

  render() {
    const backButtonComponent = this.props.backButton ? (
      <Button
        outline
        color="primary"
      >Back to finding love</Button>
    ) : null;

    const contactComponent = this.props.matchDate ? (
      <p className="contact detail">{'Phone: 111-222-3333'}</p>
    ) : null;

    const matchDateComponent = this.props.matchDate ? (
      <div className="matchDate">
        <p>{this.props.matchDate ? this.formatMatchDate() : 'A lifetime ago.'}</p>
      </div>
    ) : null;

    const matchBtns = this.props.matchBtns ? (
      <div className="matchButtons">
        <Button
          className="not"
          color="danger"
          size="lg"
          onClick={this.props.matchBtns.handlePass}
        > NOT </Button>
        <Button
          className="hot"
          color="success"
          size="lg"
          onClick={this.props.matchBtns.handleLike}
        > HOT </Button>
      </div>
    ) : null;

    return (
      <div className="UserDetail">
        {backButtonComponent}

        <img
          src={this.props.img}
          alt=""
        />
        <div className="header">
          <h2 className="name">{this.props.name}</h2>
          <h2 className="age">{this.props.age}</h2>
        </div>

        <div className="details">
          <p className="bio detail">{this.props.bio}</p>
          {contactComponent}
        </div>

        {matchBtns}

        {matchDateComponent}
      </div>
    );
  }
}

export default UserDetail;
