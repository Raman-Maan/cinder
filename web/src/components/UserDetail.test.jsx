import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UserDetail from './UserDetail';

describe('<UserDetail />', () => {
  const testData = {
    userName: 'Abc Defg',
    userAge: 22,
    userGender: 'Male',
    userBio: 'I am test user with test bio',
    userPics:
      'http://markinternational.info/data/out/613/224320906-pic-of-animals.jpg',
    matchTime: new Date(2018, 2, 5, 16, 23, 15),
    matchNote: 'This is some test notes only related about this match'
  };

  it('renders without crashing', () => {
    // simple smoke test
    shallow(<UserDetail />);
  });

  it('renders with empty props correctly', () => {
    const component = renderer.create(<UserDetail />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders with props correctly', () => {
    const component = renderer
      .create(<UserDetail userDetail={testData} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
