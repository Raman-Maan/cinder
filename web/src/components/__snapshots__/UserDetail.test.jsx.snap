// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetail /> renders with empty props correctly 1`] = `
<div
  className="UserDetail"
>
  <img
    alt=""
    src={undefined}
  />
  <div
    className="header"
  >
    <h2
      className="name"
    />
    <h2
      className="age"
    />
  </div>
  <div
    className="details"
  >
    <p
      className="bio detail"
    />
  </div>
</div>
`;

exports[`<UserDetail /> renders with props correctly 1`] = `
<div
  className="UserDetail"
>
  <img
    alt=""
    src={undefined}
  />
  <div
    className="header"
  >
    <h2
      className="name"
    />
    <h2
      className="age"
    />
  </div>
  <div
    className="details"
  >
    <p
      className="bio detail"
    />
  </div>
</div>
`;
